mok.c

```diff
@@ -223,11 +223,26 @@ EFI_STATUS import_mok_state(EFI_HANDLE image_handle)
 		UINT32 attrs = 0;
 		BOOLEAN delete = FALSE, present, addend;
 
+		addend = (v->addend_source && v->addend_size &&
+			  *v->addend_source && *v->addend_size)
+			? TRUE : FALSE;
+
 		efi_status = get_variable_attr(v->name,
 					       &v->data, &v->data_size,
 					       *v->guid, &attrs);
-		if (efi_status == EFI_NOT_FOUND)
+		if (efi_status == EFI_NOT_FOUND) {
+			if (v->rtname && addend) {
```

Comment:
Is is ok to count on the value of get_variable_attr() output parameters on an error return?
Specifically, for an EFI_NOT_FOUND error return, v->data_size won't be touched as the code is currently. And it's 0 on input because it's in a static struct. However, the value in v->data_size is used by mirror_one_mok_variable(). So I'm thinking it might be safer to guard against future changes and to make this assumption explicit by clearing it on this special case path.

`v->data_size = 0;`

Reply:
https://github.com/rhboot/shim/blob/master/lib/variables.c#L207-L231

By design, get_variable_attr() sets len to 0 to query the size of the variable.

Reply:
Yes, it's completely obvious once you pointed that out. Don't know how I missed it.
The patch looks very good.